sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs

```diff
@@ -41,14 +41,14 @@ internal ChatThreadClient(string threadId, Uri endpointUrl, CommunicationUserCre
             Id = threadId;
             _clientDiagnostics = new ClientDiagnostics(options);
             HttpPipeline pipeline = CreatePipelineFromOptions(options, communicationUserCredential);
-            _chatRestClient = new ChatRestClient(_clientDiagnostics, pipeline, endpointUrl.AbsoluteUri, options.ApiVersion);
+            _chatThreadRestClient = new ChatThreadRestClient(_clientDiagnostics, pipeline, endpointUrl.AbsoluteUri, options.ApiVersion);
         }
 
         /// <summary>Initializes a new instance of <see cref="ChatThreadClient"/> for mocking.</summary>
         protected ChatThreadClient()
         {
             _clientDiagnostics = null!;
-            _chatRestClient = null!;
```

Comment:
just noting that most libraries do not have nullable references enabled. Is there a reason for enabling it for Communication?

Reply:
no really, other than we followed the other projects on the solution.
Is there a reason NOT to enable it ? I thought this was the suggested practice in general to determine possible 'object not reference exceptions' with static analysis, is not it >

Reply:
I will defer to @tg-msft and @pakrym for what the best practice is here.

Reply:
We don't, currently, enable nullable reference types for SDK libraries because it affects public API and breaking change guidance.